### PR TITLE
Drop players who leave the server

### PR DESCRIPTION
**Pull request by AlphaKretin**
## Description

Resolves consistency issues where hosts are no longer able to mention them for various commands. In conjunction with #137, will also concede any open matches the user has.

## Checklist

- [x] I am following the [contributing guidelines]( https://github.com/AlphaKretin/emcee-tournament-bot/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: src/database/postgres.ts [hunk 1]
```diff
@@ -205,6 +205,12 @@ export class DatabaseWrapperPostgres {
 		return list.filter(p => p.tournament.status === TournamentStatus.PREPARING).map(p => this.wrap(p.tournament));
 	}
 
+	async getServerTournaments(serverId: string): Promise<DatabaseTournament[]> {
+		const list = await ChallongeTournament.find({ where: { owningDiscordServer: serverId } });
```
Comment (AlphaKretin) [inline src/database/postgres.ts:209]: I'm not sure if I need to take extra steps here to filter for non-finished tournaments.

**Comment by AlphaKretin** [comment]
Of note, this doesn't handle a user who leaves while Emcee is offline. Should a check that all players are in the appropriate server be part of the initialisation routine? That seems like a lot of overhead.

File: src/TournamentManager.ts [hunk 2]
```diff
@@ -388,6 +389,16 @@ export class TournamentManager implements TournamentInterface {
 		await this.database.cleanRegistration(msg.channelId, msg.id);
 	}
 
+	public async cleanPlayer(serverId: string, userId: string): Promise<void> {
+		// TODO: handle tournaments with user as host? we epxect more rational behaviour from hosts than players
```
Comment (kevinlul) [inline src/TournamentManager.ts:393]: typo

**Review by kevinlul — COMMENTED** [review]
I think we need a custom query for this functionality. As for the offline issue, we can either check in the startup routine or do lazy checks when some command is invoked in relation to a tournament that requires consistency in the participant list (e.g. on start, on round advance, on a score submission for that pair)

**Comment by AlphaKretin** [comment]
> I think we need a custom query for this functionality.  

I'm not sure what this would involve, could you implement it? @kevinlul 
> As for the offline issue, ...  

I wonder if we could use a utility function to get a player in any situation where them leaving the server would matter and if so drop them instead of returning?

File: src/database/postgres.ts (hunk 1 above)
Comment (kevinlul) [reply src/database/postgres.ts:209]: Yes, it's needed. Given the additional baggage here I think I should extract the events (messageCreate, guildMemberRemove) in a small refactor before we commit to more of it.

**Pull request closed without merging**